### PR TITLE
DOCSP-44006-mongosync-behavior-converting-legacy-indexes

### PR DESCRIPTION
**Pull request by amalhotra-mdb**
## DESCRIPTION
Mongosync rewrites any legacy index key values that is 0 or an empty string from the source to a valid value (1) on the destination. This should be documented on the [mongosync behavior page](https://www.mongodb.com/docs/cluster-to-cluster-sync/current/reference/mongosync/mongosync-behavior).

## STAGING
https://deploy-preview-442--docs-cluster-to-cluster-sync.netlify.app/reference/mongosync/mongosync-behavior/#legacy-index-value-rewrites

## JIRA
https://jira.mongodb.org/browse/DOCSP-44006

## SELF-REVIEW CHECKLIST

- [ ] Does each file have 3-5 taxonomy facet tags?
  See the [taxonomy tagging instructions](https://wiki.corp.mongodb.com/display/DE/Taxonomy+tagging+instructions) and this [example PR](https://github.com/10gen/cloud-docs/pull/5042/files) 
- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## EXTERNAL REVIEW REQUIREMENTS

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/reference/mongosync/mongosync-behavior.txt [hunk 1]
```diff
@@ -133,6 +133,13 @@ Read Preference
 connecting to the source and destination clusters. For more information,
 see :ref:`connections-read-preference`. 
 
+Rewrites
```
Comment (mdb-ashley) [inline source/reference/mongosync/mongosync-behavior.txt:136]: [question] Is this the right title? Is this what would help a user understand what is being rewritten?

File: source/reference/mongosync/mongosync-behavior.txt [hunk 2]
```diff
@@ -133,6 +133,13 @@ Read Preference
...
+~~~~~~~~
+
+``mongosync`` rewrites any legacy index values from the source to ``1``
```
Comment (mdb-ashley) [inline source/reference/mongosync/mongosync-behavior.txt:139]: Perhaps, 

"``mongosync`` does not support legacy index values, like ``0`` or an empty
string. Legacy index values from the source are rewritten to ``1``
on the destination."

**Review by mdb-ashley — CHANGES_REQUESTED** [review]
Thanks, Anika! Let me know when this is ready for another look.

File: source/reference/mongosync/mongosync-behavior.txt (hunk 1 above)
Comment (amalhotra-mdb) [reply source/reference/mongosync/mongosync-behavior.txt]: Thank you for taking a look at this, @mdb-ashley !

I specified the title a bit – let me know if you think it's clearer! Also, I was a little unsure if this was the write section to put this information in. Does it seem right to you, or do you think there's a more appropriate place for it?

**Comment by amalhotra-mdb** [comment]
> Thanks, Anika! Let me know when this is ready for another look.

Thank you for your review, @mdb-ashley ! BTY!

**Review by mdb-ashley — APPROVED** [review]
LGTM! Thanks, Anika!

File: source/reference/mongosync/mongosync-behavior.txt [hunk 3]
```diff
@@ -133,6 +133,13 @@ Read Preference
 connecting to the source and destination clusters. For more information,
 see :ref:`connections-read-preference`. 
 
+Legacy Index Value Rewrites
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``mongosync`` does not support legacy index values, like ``0`` or an empty
```
Comment (tdq45gj) [inline source/reference/mongosync/mongosync-behavior.txt:139]: I feel "does not support" means that it's not allowed. But mongosync is actually handling legacy index values in this case.

**Review by tdq45gj — CHANGES_REQUESTED** [review]
Sorry, I forgot to mention in the ticket that mongosync also removes any invalid index options on the destination as part of converting legacy indexes. Could you add that to the PR as well?

**Review by tdq45gj — APPROVED** [review]
LGTM, thanks!

**Pull request merged**